internal/cmd/export.go

```diff
@@ -948,73 +960,307 @@ func gatherPostgresqlLogs(ctx context.Context,
+
+	dbPods, err := clientset.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: util.DBInstanceLabels(clusterName),
 	})
+
 	if err != nil {
 		if apierrors.IsForbidden(err) {
 			writeInfo(cmd, err.Error())
 			return nil
 		}
 		return err
 	}
-	if len(pods.Items) != 1 {
-		writeInfo(cmd, "No primary instance pod found for gathering logs")
-		return nil
+
+	if len(dbPods.Items) == 0 {
+		writeDebug(cmd, "No database instance pod found for gathering logs")
 	}
 
-	podExec, err := util.NewPodExecutor(config)
+	writeDebug(cmd, fmt.Sprintf("Found %d Pods\n", len(dbPods.Items)))
+
+	for _, pod := range dbPods.Items {
+		writeDebug(cmd, fmt.Sprintf("Pod Name is %s\n", pod.Name))
+
+		podExec, err := util.NewPodExecutor(config)
+		if err != nil {
+			return err
+		}
+
+		exec := func(stdin io.Reader, stdout, stderr io.Writer, command ...string,
+		) error {
+			return podExec(namespace, pod.Name, util.ContainerDatabase,
+				stdin, stdout, stderr, command...)
+		}
+
+		// Get Postgres Log Files
+		stdout, stderr, err := Executor(exec).listPGLogFiles(numLogs)
+		if err != nil {
+			if apierrors.IsForbidden(err) {
+				writeInfo(cmd, err.Error())
+				return nil
+			}
+			return err
+		}
+		if stderr != "" {
+			writeInfo(cmd, stderr)
+		}
+
+		logFiles := strings.Split(strings.TrimSpace(stdout), "\n")
+		for _, logFile := range logFiles {
+			var buf bytes.Buffer
+
+			stdout, stderr, err := Executor(exec).catFile(logFile)
+			if err != nil {
+				if apierrors.IsForbidden(err) {
+					writeInfo(cmd, err.Error())
+					// Continue and output errors for each log file
+					// Allow the user to see and address all issues at once
+					continue
+				}
+				return err
+			}
+
+			buf.Write([]byte(stdout))
+			if stderr != "" {
+				str := fmt.Sprintf("\nError returned: %s\n", stderr)
+				buf.Write([]byte(str))
+			}
+
+			path := clusterName + fmt.Sprintf("/logs/%s/", pod.Name) + logFile
+			if err := writeTar(tw, buf.Bytes(), path, cmd); err != nil {
+				return err
+			}
+		}
+
+		// Get Postgres Conf Files
+		stdout, stderr, err = Executor(exec).listPGConfFiles()
+		if err != nil {
+			if apierrors.IsForbidden(err) {
+				writeInfo(cmd, err.Error())
+				return nil
+			}
+			return err
+		}
+		if stderr != "" {
+			writeInfo(cmd, stderr)
+		}
+
+		logFiles = strings.Split(strings.TrimSpace(stdout), "\n")
+		for _, logFile := range logFiles {
+			var buf bytes.Buffer
+
+			stdout, stderr, err := Executor(exec).catFile(logFile)
+			if err != nil {
+				if apierrors.IsForbidden(err) {
+					writeInfo(cmd, err.Error())
+					// Continue and output errors for each log file
+					// Allow the user to see and address all issues at once
+					continue
+				}
+				return err
+			}
+
+			buf.Write([]byte(stdout))
+			if stderr != "" {
+				str := fmt.Sprintf("\nError returned: %s\n", stderr)
+				buf.Write([]byte(str))
+			}
+
+			path := clusterName + fmt.Sprintf("/logs/%s/", pod.Name) + logFile
```

Comment:
We're putting the confs in a directory called logs?

Reply:
the logs directory is serving some diverse needs right now. In this PR it contains the following kinds of data, all grouped by the originating Pod.

1. Postgres logs (file-based)
2. pgBackRest logs (file-based)
3. Postgres *.conf files
4. Container logs

I think overall the Support Export is getting different kinds of information. 

1. Info about K8s resources - this is usually the YAML representation of the resource
2. Container logs
3. Processes within the containers
4. Postgres-specific information (logs and settings)
5. Patroni-specific information (patronictl output)
6. BackRest-specific information (pgbackrest info and, hopefully soon, pgbackrest check)

Figuring out how to represent that externally is a challenge, but I think there's a natural order that can be identified. Probably grouping by Pod is the best since that's a basic unit of work within k8s. But I also avoid too much change at once. So I shoved the PG settings into logs because Postgres logs and Postgres setting are closely related.

Reply:
I haven't looked at this command too deeply in a while, so please don't take me as an authority; and I definitely see some challenges in organizing the data we capture. 

But can we change this path to 
```suggestion
			path := clusterName + fmt.Sprintf("/conf/%s/", pod.Name) + logFile
```
just for the conf files?

Reply:
I think maybe changing the path from `/logs/...` to `/logs-and-config/...` may capture the spirit here. Having too flat of a structure makes it harder to examine the data within each pod. That's the challenge I'm facing here.

Reply:
ideally, I'd rather use `/pods/...` but we already have a `pods` directory that holds the YAML for each Pod. I didn't want this PR to touch too many other things.... but I'm open to suggestions.

Reply:
I'm trying to brainstorm to see if there's a path I like more than ~`pods/<name>/logs-and-config/...` and there really isn't. What's in this directory? Logs and config files. Where do they come from? The pod named <name>.

Sounds good to me and we can always iterate if we think of something more descriptive. Thanks!

Reply:
I may have found something palatable. I went with `pods`. So now there is no top-level `logs` folder. All the file-based logs roll up under the `pods` directory with the container name as a subdir. I like that when viewing each Pods' subdir, each file has the same relative path as within the Pod. It makes it easier to reason about.

These logs/conf files follow their on-disk structure (i.e. within `pgdata` or equivalent)

- PG Log files: `./pods/<container-name>/pgdata/pgNN/log/`
- PG Conf files: `./pods/<container-name>/pgdata/pgNN/`
- DB Backrest Log files: `./pods/<container-name>/pgdata/pgbackrest/`
- Repo Host Log files: `./pods/<container-name>/pgbackrest/repoN/log`

The container logs all follow a similar structure: 

- Container logs: `./pods/<container-name>/containers/`

I moved the cli log file to the root dir and gave it an extension: `/crunchy_k8s_support_export_2024-08-30-190004+0000/hippo/cli.log`. The extension is nice when viewing in VSCode.